modules/registry-configuring-registry-storage-rhodf-cephrgw.adoc

```diff
@@ -70,19 +70,14 @@ $ AWS_SECRET_ACCESS_KEY=$(oc get secret -n openshift-storage rgwtest -o yaml | g
 $ oc create secret generic image-registry-private-configuration-user --from-literal=REGISTRY_STORAGE_S3_ACCESSKEY=${AWS_ACCESS_KEY_ID} --from-literal=REGISTRY_STORAGE_S3_SECRETKEY=${AWS_SECRET_ACCESS_KEY} --namespace openshift-image-registry
 ----
 
-. Create a encryption route for Ceph RGW by entering the following command:
+. Get the `buckethost` value by entering the following command:
 +
 [source,terminal]
 ----
-$ oc create route reencrypt <route_name> --service=rook-ceph-rgw-ocs-storagecluster-cephobjectstore --port=https -n openshift-storage
+$ route_host=$(oc get objectbucket $bucket_name -n openshift-storage -o=jsonpath='{.spec.endpoint.bucketHost}')
 ----
 +
-.. Get the route host by entering the following command:
-+
-[source,terminal]
-----
-$ route_host=$(oc get route <route_name> -n openshift-storage -o=jsonpath='{.spec.host}')
-----
+
```

Comment:
After get the route_host
```
$route_host=$(oc get objectbucket obc-openshift-storage-rgwbucket  -n openshift-storage -o=jsonpath='{.spec.endpoint.bucketHost}')
$echo $route_host
rook-ceph-rgw-ocs-storagecluster-cephobjectstore.openshift-storage.svc
```
And patch the new bucket to image registy, registry operator prompt x509 error, 
which certificate I should pass for `rook-ceph-rgw-ocs-storagecluster-cephobjectstore.openshift-storage.svc` in registry setting?
```
$ oc extract secret/router-certs-default  -n openshift-ingress  --confirm
$ oc create configmap image-registry-s3-bundle --from-file=ca-bundle.crt=./tls.crt  -n openshift-config
$ oc patch config.image/cluster -p '{"spec":{"managementState":"Managed","replicas":2,"storage":{"managementState":"Unmanaged","s3":{"bucket":'\"${bucket_name}\"',"region":"us-east-1","regionEndpoint":'\"https://${route_host}\"',"virtualHostedStyle":false,"encrypt":false,"trustedCA":{"name":"image-registry-s3-bundle"}}}}}' --type=merge

$oc get config.image/cluster -o yaml 
----
    - lastTransitionTime: "2023-09-11T09:02:57Z"
      message: |-
        Unable to apply resources: unable to sync storage configuration: RequestError: send request failed
        caused by: Head "https://rook-ceph-rgw-ocs-storagecluster-cephobjectstore.openshift-storage.svc/rgwbucket-4586bf6f-e1a4-47b3-971e-281cdc4d6353": tls: failed to verify certificate: x509: certificate signed by unknown authority
      reason: Error
      status: "True"
      type: Progressing
-----

```

Reply:
Let me check

Reply:
@BlaineEXE `rook-ceph-rgw-ocs-storagecluster-cephobjectstore.openshift-storage.svc` looks like a fixed internal route, do you sure it works with https format in registry operator?

Reply:
@BlaineEXE I used the http://rook-ceph-rgw-ocs-storagecluster-cephobjectstore.openshift-storage.svc without ca added in registry operator
```
oc get config.image/cluster -o jsonpath='{.spec.storage}' | jq
{
  "managementState": "Unmanaged",
  "s3": {
    "bucket": "rgwbucket-acf3513c-dd8b-41cb-9ac1-81e752a8b119",
    "encrypt": false,
    "region": "us-east-1",
    "regionEndpoint": "http://rook-ceph-rgw-ocs-storagecluster-cephobjectstore.openshift-storage.svc",
    "virtualHostedStyle": false
  }
}
```

But with this route, I can't pull image from the ceph rgw storage with error
```
  Warning  Failed          23s                kubelet            Failed to pull image "image-registry.openshift-image-registry.svc:5000/wxj/httpd-ex@sha256:c2ccc9e66f3580457026a1c404be12269234f77bb088915f2e4218386d7c4396": rpc error: code = Unknown desc = parsing image configuration: Get "http://rook-ceph-rgw-ocs-storagecluster-cephobjectstore.openshift-storage.svc/rgwbucket-acf3513c-dd8b-41cb-9ac1-81e752a8b119/docker/registry/v2/blobs/sha256/49/49c7e05a9b639cdb1253674468fc3b337b111f987449d7b9ef309490308685a7/data?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=HFKBVMOX23YTH6VAI8H0%2F20230912%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230912T091052Z&X-Amz-Expires=1200&X-Amz-SignedHeaders=host&X-Amz-Signature=0aa38f9a3871c488eca58c71a8808decae95d721c00db8775ffc86767a419393": dial tcp: lookup rook-ceph-rgw-ocs-storagecluster-cephobjectstore.openshift-storage.svc on 192.168.221.166:53: no such host
```
How could I setting the insecure route or secured route with CA ?

Reply:
Keep in mind that `rook-ceph-rgw-ocs-storagecluster-cephobjectstore.openshift-storage.svc` is only available for ODF clusters installed in "internal mode" (installed inside OpenShift). That service is unavailable for ODF clusters installed in "external mode" where the object endpoints are outside the cluster.

Directions in this doc should not assume the service exists. For clusters (internal or external) with TLS enabled, CA certificates may be needed, but that feature isn't GA yet in ODF.